persistence/provider/sql/transaction.go

```diff
@@ -12,25 +12,30 @@ import (
 type transaction struct {
 	ds     *dataStore
 	actual *sql.Tx
+	result *persistence.TransactionResult
 }
 
 // Commit applies the changes from the transaction.
-func (t *transaction) Commit(ctx context.Context) error {
+func (t *transaction) Commit(
+	ctx context.Context,
+) (*persistence.TransactionResult, error) {
 	defer t.end()
 
 	if t.ds == nil {
-		return persistence.ErrTransactionClosed
+		return nil, persistence.ErrTransactionClosed
 	}
 
 	if err := t.ds.checkOpen(); err != nil {
-		return err
+		return nil, err
 	}
 
 	if t.actual != nil {
-		return t.actual.Commit()
+		if err := t.actual.Commit(); err != nil {
+			return nil, err
+		}
```

Comment:
As per suggestion for BoltDB:

```suggestion
		return &t.result, t.actual.Commit()
```